libnopegl/src/nodes.c

```diff
@@ -747,6 +747,10 @@ static int param_add(struct ngl_node *node, const char *key, size_t nb_elems, vo
 int ngl_node_param_add_nodes(struct ngl_node *node, const char *key,
                              size_t nb_nodes, struct ngl_node **nodes)
 {
+    if (node->scene) {
+        LOG(ERROR, "the nodes graph cannot be extended after being associated with a scene");
+        return NGL_ERROR_INVALID_USAGE;
+    }
```

Comment:
Is this a new limitation introduced by this commit or by the commit that introduced the scene association ?

Reply:
It's not allowed started from `scene: check for node sharing within the scene context`, but:
1. I can't move that commit before (since the `scene` pointer is not available yet)
2. Squashing it into that commit would make the commit quite indigestible so I prefer to keep it separate; the implication being that a few sanity checks are missing for a few commits, which I think is acceptable?

I added this to the description of that commit:
```
These operations are undefined behaviour since the introduction of the
scene pointer in the commit "check for node sharing within the scene
context".
```